integration/smartbft/smartbft_test.go

```diff
@@ -1577,6 +1572,87 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
 			By("Assert success")
 			Eventually(ordererRunners[follower.Id-1].Err(), network.EventuallyTimeout).Should(gbytes.Say("Processed proposal with seq 1"))
 		})
+
+		It("smartbft setting max backoff and LeaderHeartbeatTimeout", func() {
+			networkConfig := nwo.MultiNodeSmartBFT()
+			networkConfig.Channels = nil
+			channel := "testchannel1"
+
+			network = nwo.New(networkConfig, testDir, client, StartPort(), components)
+			network.GenerateConfigTree()
+			network.Bootstrap()
+
+			var ordererRunners []*ginkgomon.Runner
+			for _, orderer := range network.Orderers {
+				runner := network.OrdererRunner(orderer)
+				runner.Command.Env = append(
+					runner.Command.Env,
+					"FABRIC_LOGGING_SPEC=orderer.consensus.smartbft=debug:grpc=debug",
+					"ORDERER_GENERAL_BACKOFF_MAXDELAY=20s",
+				)
+				ordererRunners = append(ordererRunners, runner)
+				proc := ifrit.Invoke(runner)
+				ordererProcesses = append(ordererProcesses, proc)
+				Eventually(proc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			}
+
+			peerGroupRunner, _ := peerGroupRunners(network)
+			peerProcesses = ifrit.Invoke(peerGroupRunner)
+			Eventually(peerProcesses.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			peer := network.Peer("Org1", "peer0")
+
+			joinChannel(network, channel)
+
+			By("Waiting for followers to see the leader")
+			Eventually(ordererRunners[1].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[2].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[3].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+
+			By("Joining peers to testchannel1")
+			network.JoinChannel(channel, network.Orderers[0], network.PeersWithChannel(channel)...)
+
+			By("Deploying chaincode")
+			deployChaincode(network, channel, testDir)
+
+			By("querying the chaincode")
+			sess, err := network.PeerUserSession(peer, "User1", commands.ChaincodeQuery{
+				ChannelID: channel,
+				Name:      "mycc",
+				Ctor:      `{"Args":["query","a"]}`,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+			Expect(sess).To(gbytes.Say("100"))
+
+			By("invoking the chaincode")
+			invokeQuery(network, peer, network.Orderers[1], channel, 90)
+
+			numberKill := 3
+			orderer := network.Orderers[numberKill]
+			By(fmt.Sprintf("Killing %s", orderer.Name))
+			ordererProcesses[numberKill].Signal(syscall.SIGTERM)
+			Eventually(ordererProcesses[numberKill].Wait(), network.EventuallyTimeout).Should(Receive())
+
+			time.Sleep(time.Second * 240)
```

Comment:
why are you sleeping here?

Reply:
This is the formula for calculating the time between attempts to connect to the orderer node
```go
        backoff, max := float64(bc.Config.BaseDelay), float64(bc.Config.MaxDelay)
	for backoff < max && retries > 0 {
		backoff *= bc.Config.Multiplier
		retries--
	}
	if backoff > max {
		backoff = max
	}
	// Randomize backoff delays so that if a cluster of requests start at
	// the same time, they won't operate in lockstep.
	backoff *= 1 + bc.Config.Jitter*(grpcrand.Float64()*2-1)
	if backoff < 0 {
		return 0
	}
```

Because of "Randomize backoff delays", it was difficult to select such a sleep time in the test that the error was guaranteed to occur. 
Experimentally, I found that the leader sleeps for 2 minutes between attempts to join the 3rd order.  And during these 2 minutes, the 3rd orderer manages to get up and not wait for a heartbeat from the leader for 1 minute

Reply:
If I remove or reduce the slip, the test will succeed in the current main.

Let's say my changes aren't there. Then this test will work like this:

| event | time | delta time | 
|---|---|---|
| Killing orderer4 ($\textcolor{green}{\textsf{START TIMEOUT}}$) | 0s | 0s |
| orderer1 disconnect from orderer4 | 0s | 0s |
| orderer1 attempt connect (first backoff) | 24s | 24s |
| orderer1 attempt connect | 31s | 7s |
| orderer1 attempt connect | 38s | 7s |
| orderer1 attempt connect | 45s | 7s |
| orderer1 attempt connect | 52s | 7s |
| orderer1 attempt connect | 64s | 12s |
| orderer1 attempt connect | 83s | 19s |
| orderer1 attempt connect | 115s | 32s |
| orderer1 attempt connect | 160s | 45s |
| orderer1 attempt connect | 240s | 80s |
| Launching orderer4 ($\textcolor{green}{\textsf{STOP TIMEOUT}}$) | 241s | 1s |
| orderer4 start | 242s | 1s |
| orderer4 heartbeat message waiting - begin | 246s | 1s |
| orderer4 heartbeat message waiting - end ($\textcolor{red}{\textsf{FAIL TEST}}$) | 306s | 60s |
| orderer1 attempt connect | 361s | 55s |

Reply:
@C0rWin Through customization, I reduced the sleep time to 60 seconds

Reply:
> Experimentally, I found that the leader sleeps for 2 minutes between attempts to join the 3rd order. And during these 2 minutes, the 3rd orderer manages to get up and not wait for a heartbeat from the leader for 1 minute

Well, the problem I have with such a statement is that you are tested it locally, while CI runners might be affected differently, and we will end up adjusting this sleep interval over time.

Reply:
Yes you are right, but besides testing locally, I also tested in my fork in pipelines.
And it doesn't guarantee that this test will fail over time without my changes. Time is a delicate matter.
But right now this test shows exactly that you need to be able to manage backoff. 
And I also recommend that if LeaderHeartbeatTimeout is 1m, then MaxDelay backoff should be 20s.